dm/checker/checker.go

```diff
@@ -243,13 +245,37 @@ func (c *Checker) Init(ctx context.Context) (err error) {
 		}
 	}
 
-	if checkingShard {
-		for name, shardingSet := range sharding {
-			if shardingCounter[name] <= 1 {
-				continue
+	instance := c.instances[0]
+	if checkingShard && instance.cfg.Mode != config.ModeIncrement {
+		checkpointSQLs := []string{
+			fmt.Sprintf("SHOW CREATE TABLE `%s`.`%s`", instance.cfg.MetaSchema, cputil.LoaderCheckpoint(instance.cfg.Name)),
+			fmt.Sprintf("SHOW CREATE TABLE `%s`.`%s`", instance.cfg.MetaSchema, cputil.LightningCheckpoint(instance.cfg.Name)),
+			fmt.Sprintf("SHOW CREATE TABLE `%s`.`%s`", instance.cfg.MetaSchema, cputil.SyncerCheckpoint(instance.cfg.Name)),
+		}
+		var existCheckpoint bool
+		for _, sql := range checkpointSQLs {
+			c.tctx.Logger.Info("exec query", zap.String("sql", sql))
+			_, err := instance.targetDB.DB.QueryContext(c.tctx.Ctx, sql)
```

Comment:
how about `utils.GetTableCreateSQL` instead?

Reply:
I think they are no different. So it is ok to stay it.